src/cpp/src/llm_pipeline_static.cpp

```diff
@@ -139,49 +140,51 @@ NPULLMPipelineImpl::NPULLMPipelineImpl(
     reshape_to_static(kvcache_model, 1u, max_kvcache_size);
     // (5) Add slices to kvcache model
     kvcache_model = add_slices_to_kvcache_inputs(kvcache_model);
-    // (6) Compile both model for NPU
+    // (6) Compile both model
     m_prefill_request = core.compile_model(
-        prefill_model, "NPU", extract_config_or_empty(config, "PREFILL_CONFIG")
+        prefill_model, device, extract_config_or_empty(config, "PREFILL_CONFIG")
     ).create_infer_request();
     m_kvcache_request = core.compile_model(
-        kvcache_model, "NPU", extract_config_or_empty(config, "GENERATE_CONFIG")
+        kvcache_model, device, extract_config_or_empty(config, "GENERATE_CONFIG")
     ).create_infer_request();
     // (7) Initialize tensors
     prepare_for_new_conversation();
 };
 
-NPULLMPipelineImpl::NPULLMPipelineImpl(
+StaticLLMPipeline::StaticLLMPipeline(
     const std::filesystem::path& path,
+    const std::string& device,
     const ov::AnyMap& config
-) : NPULLMPipelineImpl(path, path.string(), config) {
+) : StaticLLMPipeline(path, path.string(), device, config) {
 }
 
-void NPULLMPipelineImpl::prepare_for_new_conversation() {
+void StaticLLMPipeline::prepare_for_new_conversation() {
     fill_tensor(m_prefill_request.get_tensor("input_ids"), m_tokenizer.get_pad_token_id());
     fill_tensor(m_prefill_request.get_tensor("position_ids"), 0u);
     fill_tensor(m_prefill_request.get_tensor("attention_mask"), 0u);
     fill_tensor(m_kvcache_request.get_tensor("attention_mask"), 0u);
     m_kvcache_desc.num_stored_tokens = 0u;
 }
 
-DecodedResults NPULLMPipelineImpl::generate(
+DecodedResults StaticLLMPipeline::generate(
     StringInputs inputs,
     OptionalGenerationConfig generation_config,
     StreamerVariant streamer
 ) {
     GenerationConfig config = (generation_config.has_value()) ? *generation_config : m_generation_config;
     if (std::holds_alternative<std::vector<std::string>>(inputs)) {
-        OPENVINO_THROW("Currently only batch size=1 is supported for NPU device");
+        OPENVINO_THROW("Currently only batch size=1 is supported");
     }
 
     OPENVINO_ASSERT(std::holds_alternative<std::string>(inputs));
-    const auto& text = std::get<std::string>(inputs);
-    auto tokenized_input = m_tokenizer.encode(text);
+    auto prompt = std::get<std::string>(inputs);
+    prompt = m_tokenizer.apply_chat_template({ {{"role", "user"}, {"content", prompt}} }, true /* add_generation_prompt */);
```

Comment:
what if chat template is not available?

Reply:
For `StatefulLLMPipeline` it's applied unconditionally under the `is_chat_conversion`. For `StaticLLMPipeline`, chat converstation isn't supported at all, so just applied w/o any checks

Reply:
is there any API from tokenizer to check it?

Reply:
@pavel-esir

Reply:
If `is_chat_conversation` is not yet supported for `StaticLLMPipeline` then apply_chat_template should not be called inside generate. If user wishes to have a chat he/she should apply template on his prompt outside of generate, like this
![image](https://github.com/openvinotoolkit/openvino.genai/assets/5703039/7183dc53-f23a-4867-8887-6553119d1a4f)

Reply:
Done